bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java

```diff
@@ -420,6 +420,23 @@ public BlackboardArtifact getGenInfoArtifact(boolean create) throws TskCoreExcep
 		return db.getMatchingArtifacts("WHERE obj_id = " + objId); //NON-NLS
 	}
 
+	@Override
+	public List<AnalysisResult> getAllAnalysisResults() throws TskCoreException {
+		return db.getBlackboard().getAnalysisResultsWhere(" obj_id = " + objId); //NON-NLS
+	}
+	
+
+	@Override
+	public List<AnalysisResult> getAnalysisResults(BlackboardArtifact.Type artifactType) throws TskCoreException {
+		
+		if (artifactType.getCategory() != BlackboardArtifact.Category.ANALYSIS_RESULT) {
+			throw new TskCoreException(String.format("Artifact type %s is not in analysis result catgeory.", artifactType.getTypeName()));
+		}
```

Comment:
I believe this sort of checking for valid arguments is important. However, invalid arguments should result in an IllegalArgumentException, not a TSKCoreException. The error is a programming error, not a problem in the TSK Core; by throwing an IllegalArgumentException, we force the developer to fix the programming error rather than log and otherwise ignore it in the TskCoreException handling.